source/Posets/Delta-V-CompletePoset.lagda

```diff
@@ -0,0 +1,854 @@
+Ian Ray, 25 July 2023.
+
+Formalizing the auxilary notion of a delta-V-complete poset which is sufficient
+for the main theorems of Section 6.2 from Tom de Jong's thesis
+  https://arxiv.org/abs/2301.12405
+  https://tdejong.com/writings/phd-thesis.pdf
+involving impredicativity (resizing axioms) in order theory.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K --exact-split #-}
+
+open import MLTT.Spartan
+open import MLTT.Two-Properties
+open import UF.FunExt
+open import UF.PropTrunc
+open import UF.Logic
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.SubtypeClassifier
+open import UF.Size
+open import UF.Equiv
+open import UF.Retracts
+open import UF.Subsingletons-FunExt
+open import UF.NotNotStablePropositions
+open import UF.Embeddings
+open import UF.Sets
+open import UF.ExcludedMiddle
+open import Slice.Family
+
+module Posets.Delta-V-CompletePoset
+ (pt : propositional-truncations-exist)
+ (fe : Fun-Ext)
+ (pe : Prop-Ext)
+  where
+
+open import Locales.Frame pt fe hiding (𝟚)
+open import Posets.TwoElementPoset pt fe
+open AllCombinators pt fe
+
+module δ-complete-poset {𝓤 𝓦 : Universe} (𝓥 : Universe) (A : Poset 𝓤 𝓦) where
+
+ δ : (x y : ∣ A ∣ₚ) → (P : Ω 𝓥) → (𝟙{𝓥} + P holds) → ∣ A ∣ₚ 
+ δ x y P (inl _) = x
+ δ x y P (inr _) = y
+
+ _≤_ : ∣ A ∣ₚ → ∣ A ∣ₚ → Ω 𝓦
+ _≤_ = rel-syntax A
+
+ open Joins (_≤_)
+
+ δ-fam : (x y : ∣ A ∣ₚ) → (P : Ω 𝓥) → Fam 𝓥 ∣ A ∣ₚ
+ δ-fam x y P = ((𝟙 + P holds) , δ x y P)
+
+ is-δ-complete : 𝓤 ⊔ 𝓦 ⊔ (𝓥 ⁺) ̇ 
+ is-δ-complete = (x y : ∣ A ∣ₚ)
+               → (x ≤ y) holds
+               → (P : Ω 𝓥)
+               → Σ s ꞉ ∣ A ∣ₚ , (s is-lub-of (δ-fam x y P)) holds
+
+ sup-of-δ : is-δ-complete
+          → (x y : ∣ A ∣ₚ)
+          → (x ≤ y) holds
+          → (P : Ω 𝓥)
+          → ∣ A ∣ₚ
+ sup-of-δ i x y o P = pr₁ (i x y o P)
+
+ module _ (i : is-δ-complete)
+          (x y : ∣ A ∣ₚ)
+          (o : (x ≤ y) holds)
+          (P : Ω 𝓥)
+           where
+
+  is-sup-of-δ : ((sup-of-δ i x y o P) is-lub-of (δ-fam x y P)) holds
+  is-sup-of-δ = pr₂ (i x y o P)
+
+  is-ub-of-δ : ((sup-of-δ i x y o P) is-an-upper-bound-of (δ-fam x y P)) holds
+  is-ub-of-δ = pr₁ is-sup-of-δ
+
+  has-lub-cond-δ : ((u , _) : upper-bound (δ-fam x y P))
+                 → ((sup-of-δ i x y o P) ≤ u) holds
+  has-lub-cond-δ = pr₂ is-sup-of-δ
+
+  lower-＝-sup-δ : ¬ (P holds) → x ＝ sup-of-δ i x y o P
+  lower-＝-sup-δ not-P = ≤-is-antisymmetric A x-≤-sup sup-≤-x
+   where
+    x-≤-sup : (x ≤ sup-of-δ i x y o P) holds
```

Comment:
I discussed this with Martin and it is preferred to give things like this "proper names", e.g. `x-below-sup`, or to just simply label them `I`, `II`, `III`, etc. Other examples in your code include `x-≤-y` and `x-≠-y`, and so on.